### PR TITLE
Get test_genomicsdbjar.sh to clone a version compatible GenomicsDB and update loader/query json files to work with protobuf

### PR DESCRIPTION
**Pull request by nalinigans**
Added GENOMICSDB_TAG to allow for compatible versions of GenomicsDB for the TestGenomicsDB classes.

Also, removing vid and callset mapping files from query.json as it already exists in loader.json and GenomicsDB throws ` ProtoBufBasedVidMapperException : Duplicate fields exist in vid map` if they are defined in both jsons

### CODE REVIEW

**Comment by mlathara** [comment]
I think we also need the minor changes from this branch: https://github.com/GenomicsDB/TestGenomicsDBJar/tree/ml_genomicsdb_release

Right now release_test uses that branch specifically...but we should really move to using the master branch of TestGenomicsDBJar. Do you want to pull those changes in here? Or just pull that PR in separately?

**Comment by nalinigans** [comment]
> I think we also need the minor changes from this branch: https://github.com/GenomicsDB/TestGenomicsDBJar/tree/ml_genomicsdb_release
> 
> Right now release_test uses that branch specifically...but we should really move to using the master branch of TestGenomicsDBJar. Do you want to pull those changes in here? Or just pull that PR in separately?

Yes, those changes are sort of incorporated in this PR. We now only need to call `run_checks.sh` when the jars are placed in the top folder as we do in `release_test.yml`. 

We also need to smoke test the published jar via `test_genomicsdbjar.sh`, but it needs to be updated to check first the staging area and then the maven repositories to get to the published jar. The published jars remain in the staging area for sometime before showing up in the maven repositories.

**Comment by nalinigans** [comment]
See [Github Actions](https://github.com/GenomicsDB/GenomicsDB/actions/runs/4570870754/jobs/8069449273) for a successful GenomicsDB release build with tag v1.5.0 with these changes.

**Comment by mlathara** [comment]
I'm not really sure I understand how this incorporates changes from the tree I pointed to: https://github.com/GenomicsDB/TestGenomicsDBJar/tree/ml_genomicsdb_release

The very minor changes there are to skip the whole bit where we download a published jar from sonatype -- instead the idea is that if the GENOMICSDB_VERSION env var is set, then we try to test using that jar (and assume that jar already exists). We use that work flow when we are doing the release process because we want to test the jar before we publish to sonatype.

I'm not sure we need to smoke test published jars anymore if the smoke test happens before we publish, right?

**Comment by nalinigans** [comment]
> I'm not really sure I understand how this incorporates changes from the tree I pointed to: https://github.com/GenomicsDB/TestGenomicsDBJar/tree/ml_genomicsdb_release
> 
> The very minor changes there are to skip the whole bit where we download a published jar from sonatype -- instead the idea is that if the GENOMICSDB_VERSION env var is set, then we try to test using that jar (and assume that jar already exists). We use that work flow when we are doing the release process because we want to test the jar before we publish to sonatype.

I went down the path of invoking `run_checks.sh` before realizing you had made the change in `test_genomicsdbjar.sh` in the `ml_genomicsdb_release` branch. Instead of bypassing the downloading part in `test_genomicsdbjar.sh`, I am invoking the run_checks.sh directly, thats really the only change. But, no problem, I can pull in your changes and continue to call `test_genomicsdbjar.sh` from `release_test.yml`. 

I do have a PR - https://github.com/GenomicsDB/GenomicsDB/pull/282 reflecting the GenomicsDB branch/tag we want to use for the TestGenomicsDB* classes, this is important when the classes in `master` and the branch we are testing with are incompatible.
.
> 
> I'm not sure we need to smoke test published jars anymore if the smoke test happens before we publish, right?

Yeah, really not necessary, the mvn checksum verification should take care of the integrity of the jars.

**Comment by nalinigans** [comment]
We should have had merged [ml_genomicsdb_release](https://github.com/GenomicsDB/TestGenomicsDBJar/tree/ml_genomicsdb_release) to `master` in hindsight...

**Comment by nalinigans** [comment]
@mlathara, I have merged `ml_genomicsdb_release`. Can you please review?

File: test_genomicsdbjar.sh [hunk 1]
```diff
@@ -43,8 +46,10 @@ fi
 echo "Done"
 echo
 
+GENOMICSDB_TAG=${GENOMICSDB_TAG:-v$GENOMICSDB_VERSION}
```
Comment (mlathara) [inline test_genomicsdbjar.sh:49]: minor: I guess we only need one of `GENOMICSDB_TAG` or `GENOMICSDB_VERSION` not both here...?

**Review by mlathara — APPROVED** [review]
only a minor comment. sorry, I totally spaced on reviewing this.

File: test_genomicsdbjar.sh (hunk 1 above)
Comment (nalinigans) [reply test_genomicsdbjar.sh:49]: I would like to keep `GENOMICSDB_TAG` separate from `GENOMICSDB_VERSION` to check for backward compatibility. That way, we will be required to bump the major/minor version when the backward compatibility is broken. Can augment this PR - https://github.com/GenomicsDB/GenomicsDB/pull/282 with a backward compatibility check. Does that sound OK?

**Pull request merged**